includes/enrolment/class-sensei-course-enrolment.php

```diff
@@ -29,6 +29,7 @@
 class Sensei_Course_Enrolment {
 	const META_PREFIX_ENROLMENT_RESULTS = 'sensei_course_enrolment_';
 	const META_COURSE_ENROLMENT_VERSION = '_course_enrolment_version';
+	const META_PREFIX_REMOVED_LEARNER   = 'sensei_removed_learner_';
```

Comment:
Maybe we should consider adding a single meta `sensei_removed_learners` per course?

We could then store a JSON object with the following format:
```
{
	<user_id> : {
		date: The date of removal
		reason: The reason of removal
	}
	<another_user_id> : ...
}
```

This way we could use the same functionality for other reasons in the future and we could provide more information on the user about blocking.

Reply:
Hey @gkaragia ! I like the idea to have a single meta per course, but as we can't use the JSON_EXTRACT because the support, I think it can make hard some queries when we want in the future, and also we can't use de WP funcions to these queries too (just get the data and parse separately). WDTY about that?

Reply:
After talking with @gkaragia, besides the JSON idea, he also talked about removing the term when we're removing a user. So it can help us with some queries (what makes store it in JSON doesn't be a problem), I did it by calling the `save_enrolment`. The commit is here: https://github.com/Automattic/sensei/pull/3402/commits/6b154d43ced496d35b4348fd35809869b74aecc9
Thank you for the suggestions, @gkaragia ! WDYT about the refactored code?

I just didn't add the `reason` for now, because it will always happen from the way. But if we have different ways in the future, or we offer a text field for the user, we can add this property to the object.